service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java

```diff
@@ -222,18 +223,19 @@ public HoldingsRecord getHolding(String instanceId, String holdingId) {
                 loanIdToLoan.put(loanId, loan);
             }
 
-            for (JsonNode instance : getInstances(instanceIdToPartialLoan.keySet())) {
+            for (JsonNode instance : getInstances(instanceIdToLoanId.keySet())) {
                 String instanceId = getText(instance, "/id");
                 instanceIdToInstance.put(instanceId, instance);
             }
 
-            for (JsonNode item : getItems(itemIdToPartialLoan.keySet())) {
+            for (JsonNode item : getItems(itemIdToLoanId.keySet())) {
                 String itemId = getText(item, "/id");
                 itemIdToItem.put(itemId, item);
             }
 
-            for (JsonNode partialLoan : instanceIdToPartialLoan.values()) {
-                String loanId = getText(partialLoan, "/id");
+            for (String loanId : itemIdToLoanId.values()) {
+                JsonNode partialLoan = loanIdToPartialLoan.get(loanId);
```

Comment:
This map lookups make this a less performant change. May be negligible as is the changing pointers in the map.